### ISSUE TITLE
Automate tax deductible emails

### ISSUE THREAD

**Issue by BlueHat-GURU** [magfest/ubersystem#2426]
Include an automated email with the amount of extra donation that people kicked in for the calendar year.

Probably depends on persistent uber.

Might also need some calculations regarding swag vs fair market value for supporters.

@nickthenewbie1 can provide more details

**Comment by binary1230** [comment]
This task has a lot of subtle backend stuff needed, very precise tax info
is required to be known.  It is something we need later this year, along
with some reporting along the same lines.   Probably doesn't depend on
persistence

**Comment by bds002** [comment]
Yeah as long as all applicable values are defined, the email can be sent
out from individual instances.

**Comment by binary1230** [comment]
(in front of computer now) more details:

This issue describes one of two related things that we need, by the end of each calendar year (ending Dec 31).

1. The feature this github issue is talking about is: an automated email sent to each attendee who donated money above a certain threshold (I think $250, need to check).  this serves as a non-profit donation receipt which has to have a very specific format, very specific wording from the IRS, and the money laid out in a very specific way.
2. Related feature that we also need:  https://github.com/magfest/ubersystem/issues/1919 This is a report, used by the accountants on our real taxes, which breaks down purchases made through ubersystem by some of the following ways:  base price of badge, fair market value of swag, amount extra they paid over the fair market value.  maybe other stuff.

BOTH of those tasks, from a technical standpoint, are pretty easy.  What is challenging is that they require probably a rewrite of the underlying data structures we use to input badge prices in ubersystem.  

For example, the existing way we describe a supporter badge in the INI is:
- $180 -> supporter pack

And the new way might have to be something like:
supporter pack:
- $80 base badge price
- $40 fair market value  of the items you get in the supporter pack
- $40 leftover is a tax deductible donation to magfest

(I have some whiteboard chickenscratch and notes about how we might structure this data, when we're ready to work on this, hit me up and I will post it here [need to do some digging to find it])

What is also challenging is that, normally, when we open preregistration, we don't exactly know WHAT is in the supporter pack, only that there will be some random swag.  i.e. at reg launch, we don't know that $40 is the fair market value.  We won't know until we actually buy the stuff and negotiate with suppliers.

So changing up how this works might require us to do stuff like:
- using the "fair market value" as a $40 budget when buying swag
- be flexible enough to allow us to leave blank, or change, the fair market value later in the year when we are able to calculate it.  this would mean we don't send the emails out until later in the year

potentially related to: 
- https://github.com/magfest/ubersystem/issues/2070
- https://github.com/magfest/ubersystem/issues/2069


One other note:  I would categorize this task as critical/important (as Nick can attest having done it manually), but also, it's not as high a priority til later in the year.  Still, we do tend to get busy later in the year, so it would be good to find out when deadlines are needed for 1) sending emails to attendees for this and 2) when accounting (Steph and co) need the report.

**Comment by binary1230** [comment]
btw also, completing this task -might- (I'm not sure, @kitsuta is the resident expert and can say better than me) require us to do some refactoring of all the different price-related DB fields on the Attendee and Group model, like .amount_extra and other related fields.

**Comment by binary1230** [comment]
continuing braindump:

my chickenscratch that no one should even attempt to read, putting here so i can braindump more info from it later:
![20160829_175400](https://cloud.githubusercontent.com/assets/5413064/24711004/bf042b7c-19ec-11e7-8aff-3fb939f4ab8b.jpg)
![img_20160829_175655 2](https://cloud.githubusercontent.com/assets/5413064/24711005/bf05a312-19ec-11e7-89d5-96e7f35a9289.jpg)

**Comment by binary1230** [comment]
I believe Tony have more info on the exact format of fair market vs cost of goods vs other stuff.
